Test value for null or undefined

DennisKuhn opened the pull request:
### Reasons for making this change

Previous test caused 0 to result in a blank field

[Please describe them here]

If this is related to existing tickets, include links to them as well.

https://github.com/rjsf-team/react-jsonschema-form/issues/1786

epicfaace commented on packages/material-ui/src/TextWidget/TextWidget.tsx:
```diff
@@ -43,7 +43,7 @@ const TextWidget = ({
         name={name}
         type={schema.type as string}
-        value={value ? value : ''}
+        value={value === undefined || value === null  ? '' : value}
```
I believe we would also need to include NaN. Can we do this instead (because I believe the only falsy value we need to include is 0)?

```suggestion
        value={value || (value === 0)  ? '' : value}
```

DennisKuhn commented:
Not sure if I've done it right:
- added a test for the 0 case, committed test file and snap shot file. 
- updated the code to `value || (value === 0) ? value : ''`
